examples/domain-generalization-with-kd/README.md

```diff
@@ -0,0 +1,75 @@
+This directory contains an implementation of multi-source domain generalization via knowledge distillation for SQuAD-style reading comprehension tasks, as proposed in the following EMNLP 2022 paper:
+
+```
+Not to Overfit or Underfit the Source Domains? An Empirical Study of Domain Generalization in Question Answering
+Md Arafat Sultan, Avirup Sil and Radu Florian
+```
```

Comment:
add a link to the paper? Also I think it would be helpful to very quickly explain what knowledge distillation is.